scripts/needs_ci_run

```diff
@@ -99,14 +103,20 @@ print(f'last build successful?, gitsha: {status}, {last_pr_gitsha}')
 if not status:
     sys.exit(0)
 
+# if commit does not exist anymore, dont skip
+if subprocess.run(['git', 'cat-file', '-t', 'last_pr_gitsha'],
+                  capture_output=True,
+                  text=True).stdout != 'commit':
+    sys.exit(0)
```

Comment:
wow you decided to also fix this as well? i remember recently mentioning this to @mandeepsandhu on a certain other PR but i forget if we made an issue for it

Reply:
I faced this issue when doing some force pushes and fixed it :)